typecats/tc.py

```diff
@@ -132,11 +127,32 @@ def register_unstruc_hook(*args, **kwargs):
     _TYPECATS_DEFAULT_CONVERTER.register_unstructure_hook(*args, **kwargs)
 
 
+def register_struc_hook_func(*args, **kwargs):
+    """Use this to register cattrs structuring hooks on the internal cattrs Converter"""
```

Comment:
In what cases should I specifically set a structure hook on the internal cattrs Converter?

Reply:
We already had `register_struc_hook` and `register_unstruc_hook` and I had a need for their respective `register_..._func` variations in housecats so I added those two. The difference is that the traditional method takes only maps a `type` to a hook where as `_func` allows you to provide a predicate function and a hook. The predicate function allows you to have programmatic control over whether a specific type should use the matching hook. With GenConverter, `register_..._hook` no longer works with generics (you'd have to register one for each bound type, such as `List[str]`, `List[int]` and so on rather than just `List`.